python/ray/dataframe/dataframe.py

```diff
@@ -650,36 +616,21 @@ def sum(self, axis=None, skipna=True, level=None, numeric_only=None):
         Returns:
             The sum of the DataFrame.
         """
-        if axis is not None:
-            self._row_index._get_axis_name(axis)
 
-        sum_df = pd.concat(ray.get(
+        axis = self._row_index._get_axis_name(axis) if axis is not None \
```

Comment:
This allows us to reduce the axis down to a single check for the future, which is nice.